<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Alexander J. Titus" />
  <meta name="author" content="Matthew E. Walsh" />
  <meta name="dcterms.date" content="2024-06-14" />
  <meta name="keywords" content="Artificial Intelligence, Generative AI, Machine Learning, Biology" />
  <title>Generative Biology</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="Generative Biology" />
  <meta name="citation_title" content="Generative Biology" />
  <meta property="og:title" content="Generative Biology" />
  <meta property="twitter:title" content="Generative Biology" />
  <meta name="dc.date" content="2024-06-14" />
  <meta name="citation_publication_date" content="2024-06-14" />
  <meta property="article:published_time" content="2024-06-14" />
  <meta name="dc.modified" content="2024-06-14T23:06:35+00:00" />
  <meta property="article:modified_time" content="2024-06-14T23:06:35+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Alexander J. Titus" />
  <meta name="citation_author_institution" content="In Vivo Group, Washington, DC, USA" />
  <meta name="citation_author_institution" content="International Computer Science Institute, Berkeley, CA, USA" />
  <meta name="citation_author_orcid" content="0000-0002-0145-9564" />
  <meta name="citation_author" content="Matthew E. Walsh" />
  <meta name="citation_author_institution" content="Center for Health Security, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA" />
  <meta name="citation_author_institution" content="Department of Environmental Health and Engineering, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA" />
  <meta name="citation_author_orcid" content="0000-0003-1514-7761" />
  <link rel="canonical" href="https://In-Vivo-Group.github.io/generative-biology/" />
  <meta property="og:url" content="https://In-Vivo-Group.github.io/generative-biology/" />
  <meta property="twitter:url" content="https://In-Vivo-Group.github.io/generative-biology/" />
  <meta name="citation_fulltext_html_url" content="https://In-Vivo-Group.github.io/generative-biology/" />
  <meta name="citation_pdf_url" content="https://In-Vivo-Group.github.io/generative-biology/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://In-Vivo-Group.github.io/generative-biology/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://In-Vivo-Group.github.io/generative-biology/v/4ff558bab8053e90d52344df8c4e8af9f14155af/" />
  <meta name="manubot_html_url_versioned" content="https://In-Vivo-Group.github.io/generative-biology/v/4ff558bab8053e90d52344df8c4e8af9f14155af/" />
  <meta name="manubot_pdf_url_versioned" content="https://In-Vivo-Group.github.io/generative-biology/v/4ff558bab8053e90d52344df8c4e8af9f14155af/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Generative Biology</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://In-Vivo-Group.github.io/generative-biology/v/4ff558bab8053e90d52344df8c4e8af9f14155af/">permalink</a>)
was automatically generated
from <a href="https://github.com/In-Vivo-Group/generative-biology/tree/4ff558bab8053e90d52344df8c4e8af9f14155af">In-Vivo-Group/generative-biology@4ff558b</a>
on June 14, 2024.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Alexander J. Titus</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0145-9564">0000-0002-0145-9564</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/alexandertitus">alexandertitus</a>
<br>
<small>
In Vivo Group, Washington, DC, USA; International Computer Science Institute, Berkeley, CA, USA
· Funded by Grant TBD
</small></p></li>
<li><p><strong>Matthew E. Walsh</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1514-7761">0000-0003-1514-7761</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/mwalsh52">mwalsh52</a>
<br>
<small>
Center for Health Security, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA; Department of Environmental Health and Engineering, Johns Hopkins Bloomberg School of Public Health, Baltimore, MD, USA
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/In-Vivo-Group/generative-biology/issues">GitHub Issues</a>
or email to
Alexander J. Titus &lt;publications@theinvivogroup.com&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>The rapid pace of progress in generative artificial intelligence (AI) techniques like deep learning, reinforcement learning, and transformer neural networks is transforming the life sciences and biomedicine. This living review paper provides an updatable, comprehensive overview and analysis of the latest literature on generative biology – the application of cutting-edge generative AI methods to accelerate insights and innovation across the life sciences and healthcare. All authors are welcome to contribute to this review via pull requests.</p>
<p>The review synthesizes key developments in using generative models for de novo biomedical discovery, design, and decision support. It examines techniques and applications including deep learning on omics data for personalized medicine, generative chemistry for drug development, protein structure prediction for molecular engineering, image synthesis for pathology, language models for clinical decision support, robotic simulation for prosthetics, and generative networks for cell programming.</p>
<p>The review highlights representative studies and benchmarks in each area while contextualizing progress, limitations, emerging best practices, and directions for future work. It also discusses social and ethical challenges raised by generative biology applications, such as compounding bias, system opacity, and dual-use risks, alongside proposed solutions.</p>
<p>As a living review, this paper will be continually updated as the field rapidly advances to provide researchers and practitioners with an up-to-date reference on the state of the art in employing generative AI to accelerate biomedicine for the collective good.</p>
<h1 class="page_break_before" id="executive-summary">Executive Summary</h1>
<p>The goal with be a 2 page TL;DR of the review after v1 is complete. Need more content.</p>
<h1 id="introduction">Introduction</h1>
<p>This is the start of the Generative Biology living review!</p>
<h1 id="computers-algorithms-and-the-internet">Computers, Algorithms and the Internet</h1>
<h2 id="s-and-1960s-early-computers-and-algorithms">1950s and 1960s: Early computers and algorithms</h2>
<p>Computers were used in the early 1950s for population genetics calculations <span class="citation" data-cites="gldEpQ9">[<a href="#ref-gldEpQ9" role="doc-biblioref">1</a>]</span>. Notably, the inception of computational modeling in biology dates to the origins of computer science itself. British mathematician and logician Alan Turing, often referred to as “the father of computing”, used primitive computers to implement a model of biological morphogenesis (the emergence of pattern and shape in living organisms) in 1952 <span class="citation" data-cites="UK6rEYVY">[<a href="#ref-UK6rEYVY" role="doc-biblioref">2</a>]</span>. At about the same time, a computer called MANIAC was used for measuring speculative genetic codes; it was originally built for weaponry research at the Los Alamos National Laboratory in New Mexico <span class="citation" data-cites="RqzQpRL0">[<a href="#ref-RqzQpRL0" role="doc-biblioref">3</a>]</span>.</p>
<p>Computers were used for the study of protein structure by the 1960s, and other increasingly diverse analyses. These developments marked the rise of the computational biology field, stemming from research focused on protein crystallography, in which scientists found computers indispensable for carrying out laborious Fourier analyses to determine the three-dimensional structure of proteins <span class="citation" data-cites="AtEWfwCG WmbDqOWv">[<a href="#ref-AtEWfwCG" role="doc-biblioref">4</a>,<a href="#ref-WmbDqOWv" role="doc-biblioref">5</a>]</span>.</p>
<p>In addition to advances in determination of protein structures through crystallography, the first sequence of protein, insulin, was published <span class="citation" data-cites="3t4YHcSd Htmw8erk">[<a href="#ref-3t4YHcSd" role="doc-biblioref">6</a>,<a href="#ref-Htmw8erk" role="doc-biblioref">7</a>]</span>. More efficient protein sequencing methods, such as the Edman degradation technique <span class="citation" data-cites="Dq9jMQ6v">[<a href="#ref-Dq9jMQ6v" role="doc-biblioref">8</a>]</span>, enabled sequencing 15 different proteins over a decade <span class="citation" data-cites="prCR1hpf">[<a href="#ref-prCR1hpf" role="doc-biblioref">9</a>]</span>. COMPROTEIN, one of the first bioinformatics softwares developed in the early 1960s, was designed to overcome the limitations of Edman sequencing <span class="citation" data-cites="1Bxyvu5tQ">[<a href="#ref-1Bxyvu5tQ" role="doc-biblioref">10</a>]</span>. In an effort to simplify the handling of protein sequence data for the COMPROTEIN software, a one-letter amino acid code was developed <span class="citation" data-cites="ibvKtnrl">[<a href="#ref-ibvKtnrl" role="doc-biblioref">11</a>]</span>. This one-letter code was first used in the Atlas of Protein Sequence and Structure <span class="citation" data-cites="YYcdJkZI">[<a href="#ref-YYcdJkZI" role="doc-biblioref">12</a>]</span>, the first biological sequence database, laying the groundwork for paleogenetic studies.</p>
<p>Development of methods to compare protein sequences followed. The Needleman-Wunsch algorithm <span class="citation" data-cites="C1oThCRN">[<a href="#ref-C1oThCRN" role="doc-biblioref">13</a>]</span>, the first dynamic programming algorithm developed for pairwise protein sequence alignments, was introduced in the 1970s. Multiple sequence alignment (MSA) algorithms followed in the early 1980s. Progressive sequence alignment was introduced by Feng and Doolittle in 1987 <span class="citation" data-cites="9IbNBDhy">[<a href="#ref-9IbNBDhy" role="doc-biblioref">14</a>]</span>. The MSA software CLUSTAL, a simplification of the Feng-Doolittle algorithm <span class="citation" data-cites="r9luxm5X">[<a href="#ref-r9luxm5X" role="doc-biblioref">15</a>]</span> was developed in 1988. It is still used and maintained to this day <span class="citation" data-cites="10odPMDxs">[<a href="#ref-10odPMDxs" role="doc-biblioref">16</a>]</span>.</p>
<h2 id="s-from-protein-to-dna-analysis">1970s: From protein to DNA analysis</h2>
<p>The deciphering of all 64 triplet codons of the genetic code in 196817 fueled a desire to efficiently determine the sequence of DNA that existed into the 1970s. This desire led to the development of cost-efficient DNA sequencing methods, such as the Maxam-Gilbert and Sanger sequencing techniques in the mid-1970s <span class="citation" data-cites="NQbXc701 3t4YHcSd Htmw8erk">[<a href="#ref-3t4YHcSd" role="doc-biblioref">6</a>,<a href="#ref-Htmw8erk" role="doc-biblioref">7</a>,<a href="#ref-NQbXc701" role="doc-biblioref">17</a>]</span>. With this new ability to generate DNA sequence data, a paradigm shift from protein analysis to DNA analysis occurred in the late 1970s. Concurrently, concerns over recombinant DNA research led to safety protocols established during the 1975 Asilomar conference <span class="citation" data-cites="10pibmvLC">[<a href="#ref-10pibmvLC" role="doc-biblioref">18</a>]</span>.</p>
<p>New DNA sequencing techniques resulted in significantly more data to be analyzed, a task at which computation could help. The first software dedicated to analyzing Sanger sequencing reads was published in 1979 <span class="citation" data-cites="M8aKGlSe">[<a href="#ref-M8aKGlSe" role="doc-biblioref">19</a>]</span>. DNA sequences began to be utilized in phylogenetic inference with pioneering methods like maximum likelihood for inferring phylogenetic trees from DNA sequences <span class="citation" data-cites="nzs4mjEF">[<a href="#ref-nzs4mjEF" role="doc-biblioref">20</a>]</span>. Several bioinformatics tools and statistical methods were developed following this work. The adoption of Bayesian statistics in molecular phylogeny in the 1990s was inspired by this <span class="citation" data-cites="1CARqSM3b">[<a href="#ref-1CARqSM3b" role="doc-biblioref">21</a>]</span> and is still commonly used in biology today <span class="citation" data-cites="17nhyKsKm">[<a href="#ref-17nhyKsKm" role="doc-biblioref">22</a>]</span>. Yet, numerous computational limitations needed to be overcome during the latter half of the 1970s to expand the utilization of computing in the life sciences, especially in DNA analysis. The subsequent decade proved instrumental in addressing these challenges.</p>
<figure>
<img src="https://github.com/samadon1/generative-biology/assets/56901167/4d765e41-8a46-4eb8-9b45-c206e0ab22e3" alt="Figure 1: The history of parallel advancements in computing and the life sciences: A timeline of major milestones." />
<figcaption aria-hidden="true"><strong>Figure 1</strong>: The history of parallel advancements in computing and the life sciences: A timeline of major milestones.</figcaption>
</figure>
<h2 id="s-simultaneous-advances-in-computing-and-biology">1980s: Simultaneous advances in computing and biology</h2>
<p>Parallel advancements in biology and computing propelled bioinformatics forward during the 1980s and 1990s. Molecular techniques like gene targeting and amplification, using enzymes like restriction endonucleases and DNA ligases, laid the groundwork for genetic engineering <span class="citation" data-cites="10pibmvLC">[<a href="#ref-10pibmvLC" role="doc-biblioref">18</a>]</span>. The polymerase chain reaction (PCR) transformed gene amplification, while innovations like Taq polymerase and thermal cyclers optimized the process <span class="citation" data-cites="CvVpaf4a">[<a href="#ref-CvVpaf4a" role="doc-biblioref">23</a>]</span>.</p>
<p>Computing accessibility surged with microcomputers like the Commodore PET, Apple II, and Tandy TRS-80, along with bioinformatics software like the GCG software suite <span class="citation" data-cites="12FKql4zv">[<a href="#ref-12FKql4zv" role="doc-biblioref">24</a>]</span> and DNASTAR <span class="citation" data-cites="17EunIaqv">[<a href="#ref-17EunIaqv" role="doc-biblioref">25</a>]</span>, another sequence manipulation suite capable of assembling and analyzing Sanger sequencing data. Other sequence manipulation suites were developed to run on CP/M, Apple II, and Macintosh computers <span class="citation" data-cites="dSNqAXK0">[<a href="#ref-dSNqAXK0" role="doc-biblioref">26</a>]</span> in the years 1984 and 1985. Free code copies of this software were offered on demand by some developers. This propelled an upcoming software-sharing movement in the programming world <span class="citation" data-cites="17gPkdn86 WD5AIwcy">[<a href="#ref-17gPkdn86" role="doc-biblioref">27</a>,<a href="#ref-WD5AIwcy" role="doc-biblioref">28</a>]</span>.</p>
<p>The free software movement, led by the GNU project, promoted open-source bioinformatics tools. Major sequence databases (EMBL, GenBank, DDBJ) standardized data formatting and enabled global sharing. Bioinformatics journals, like CABIOS, which is now known as Bioinformatics (Oxford, England) accentuated computational methods’ importance. Desktop workstations with Unix-like systems and scripting languages aided bioinformatics analyses, and scripting languages simplified tool development.</p>
<h2 id="s-the-genomics-era-and-web-based-bioinformatics">1990s: The genomics era and web-based bioinformatics</h2>
<p>The genomics era began in the mid-1990s with the complete sequencing of the Haemophilus influenzae genome <span class="citation" data-cites="11GoDnHp3">[<a href="#ref-11GoDnHp3" role="doc-biblioref">29</a>]</span>, initiating genome-scale analyses. This milestone was followed by the publication of the human genome at the beginning of the 21st century, which served as the definitive catalyst for the genomic era <span class="citation" data-cites="7GkBaAWF">[<a href="#ref-7GkBaAWF" role="doc-biblioref">30</a>]</span>. This transformative event spurred the design and development of several specialized Perl-based software to assemble whole-genome sequencing reads: PHRAP <span class="citation" data-cites="Ero6Muzo">[<a href="#ref-Ero6Muzo" role="doc-biblioref">31</a>]</span>, Celera Assembler <span class="citation" data-cites="18guJvxxN">[<a href="#ref-18guJvxxN" role="doc-biblioref">32</a>]</span> among others.</p>
<p>Tim Berners-Lee’s pioneering work at CERN in the early 1990s resulted in the World Wide Web, transforming global communication and ushering in an era of unprecedented access to information. With the advent of the internet, researchers gained a powerful platform to share and access vast amounts of biological data efficiently. This facilitated collaborative efforts in biology and genomics, leading to the establishment of foundational databases such as the EMBL Nucleotide Sequence Data Library <span class="citation" data-cites="i2bVokif">[<a href="#ref-i2bVokif" role="doc-biblioref">33</a>]</span> and the GenBank database became the responsibility of the NCBI <span class="citation" data-cites="Tx5CQPm8">[<a href="#ref-Tx5CQPm8" role="doc-biblioref">34</a>]</span> in 1992. Also, the famous NCBI website came online in 1994, featuring the efficient pairwise alignment tool BLAST <span class="citation" data-cites="qPG24gP6">[<a href="#ref-qPG24gP6" role="doc-biblioref">35</a>]</span>. After that, the world saw the birth of major databases we still rely on today: Genomes (1995), PubMed (1997), and Human Genome (1999) <span class="citation" data-cites="14cnpLsxs 1DNwvIpHH yOdF9F1J">[<a href="#ref-14cnpLsxs" role="doc-biblioref">36</a>,<a href="#ref-1DNwvIpHH" role="doc-biblioref">37</a>,<a href="#ref-yOdF9F1J" role="doc-biblioref">38</a>]</span>.</p>
<p>The proliferation of web-based resources transformed access to bioinformatics tools, democratizing their availability and usability for researchers worldwide. Through the development of web platforms, bioinformatics tools became more user-friendly and accessible. This shift enabled researchers to interact with sophisticated analytical tools without needing extensive computational expertise or access to specialized hardware. Consequently, the widespread adoption of web-based bioinformatics resources facilitated broader participation in genomic and molecular research, accelerating scientific discovery and collaboration on a global scale. Graphical web servers emerged as a convenient alternative to traditional UNIX-based systems, simplifying data analysis without the need for complex installations. The continued relevance of servers for scientific purposes is exemplified by the AlphaFold Server which uses the latest AlphaFold 3 model <span class="citation" data-cites="MDJLJlU">[<a href="#ref-MDJLJlU" role="doc-biblioref">39</a>, J. et al. Accurate structure prediction of biomolecular interactions with AlphaFold 3. Nature 1–3 (2024) doi:10.1038/s41586-024-07487-w]</span>, released in 2024, to provide highly accurate biomolecular structure predictions in a unified platform.</p>
<p>The internet facilitated the dissemination of scientific research through online publications, challenging traditional print-based methods. Early initiatives like BLEND <span class="citation" data-cites="UglDtG3f">[<a href="#ref-UglDtG3f" role="doc-biblioref">40</a>]</span> paved the way for internet-based scientific publishing by shedding insights into the potentials and obstacles associated with using the internet for scientific publications. This study paved the way for leveraging the Internet for both data set storage and dissemination, leading up to the establishment of preprint servers like arXiv (est. 1991) <span class="citation" data-cites="jMSKwR4o">[<a href="#ref-jMSKwR4o" role="doc-biblioref">41</a>]</span> and bioRxiv (est. 2013) <span class="citation" data-cites="iWYRgRGv">[<a href="#ref-iWYRgRGv" role="doc-biblioref">42</a>]</span> which changed the way scientific findings are shared and accessed. These platforms democratized access to scientific knowledge by enabling researchers to share their work rapidly and openly, facilitating interdisciplinary collaborations and the cross-pollination of ideas.</p>
<p>The experimental determination of the first three-dimensional structure of a protein, specifically, myoglobin, occurred in 1958 via X-ray diffraction <span class="citation" data-cites="AtEWfwCG">[<a href="#ref-AtEWfwCG" role="doc-biblioref">4</a>]</span>. However, earlier groundwork by Pauling and Corey with the publication of two articles in 1951 that reported the prediction of α-helices and β-sheets <span class="citation" data-cites="4gIVWSCR">[<a href="#ref-4gIVWSCR" role="doc-biblioref">43</a>]</span> laid the foundation for predicting protein structures. Similar to advances in other biological sciences, the utilization of computers has made it feasible to conduct calculations aimed at predicting the secondary and tertiary structure of proteins, with varying levels of confidence. This capability has been notably enhanced by the development of fold recognition algorithms, also known as threading algorithms <span class="citation" data-cites="cc2nygki eM1kmLmc">[<a href="#ref-cc2nygki" role="doc-biblioref">44</a>,<a href="#ref-eM1kmLmc" role="doc-biblioref">45</a>]</span>. However, proteins are dynamic entities, requiring advanced biophysical models to describe their interactions and movements accurately. Force fields have been formulated to describe the interactions among atoms, enabling the introduction of tools for modeling the molecular dynamics of proteins during the 1990s <span class="citation" data-cites="Z7e2T8go">[<a href="#ref-Z7e2T8go" role="doc-biblioref">46</a>]</span>. Used to study the behavior and interactions of atoms and molecules over time, molecular dynamics simulations calculate the positions and velocities of atoms based on physical principles. Despite the theoretical advancements and availability of tools, executing molecular dynamics simulations remained challenging in practice due to the substantial computational resources they demanded.</p>
<p>Graphical processing Units (GPUs) have made molecular dynamics more accessible <span class="citation" data-cites="sT4CEJLS">[<a href="#ref-sT4CEJLS" role="doc-biblioref">47</a>]</span>, with applications extending to other bioinformatics fields requiring intensive computation. However, the internet’s role in data dissemination, coupled with increasing computational power, has led to the proliferation of ‘Big Data’ in bioinformatics.</p>
<h2 id="high-throughput-sequencing-and-big-data">2000-2010: High-throughput sequencing and big data</h2>
<p>Second-generation sequencing technologies democratized high-throughput bioinformatics. For example ‘454’ pyrosequencing, a high-throughput DNA sequencing technique played a significant role in advancing genomics research by enabling rapid and cost-effective sequencing of DNA samples, particularly for applications such as whole-genome sequencing <span class="citation" data-cites="16JwdMPBH">[<a href="#ref-16JwdMPBH" role="doc-biblioref">48</a>]</span>, but computational challenges arose with increased data volumes. Decreasing sequencing costs resulted in more data being generated, emphasizing data organization and accessibility. Specialized repositories and standardization efforts were needed to ensure data interoperability. High-performance computing adaptation became vital to address the increased amounts of data within bioinformatics projects. The surge in bioinformatics projects, accompanied by a vast influx of data, prompted adjustments from funding bodies to accommodate the demand for high-performance computing resources and collaborative initiatives.</p>
<p>While basic computer setups suffice for some projects, others demand complex infrastructures and substantial expertise. Government-sponsored entities like <a href="https://computecanada.ca/">Compute Canada</a>, <a href="https://esd.ny.gov/doing-business-ny/new-york-state-high-performance-computing-program">New York State’s High-Performance Computing Program</a>, <a href="https://www.etp4hpc.eu/">The European Technology Platform for High-Performance Computing</a>, and <a href="https://www.nchc.org.tw/?langid=2">National Center for High-Performance Computing</a> served researchers’ computational needs. Companies like Amazon, Microsoft, and Google, among many others, offer bioinformatics and life sciences services, emphasizing the field’s importance.</p>
<h3 id="table-1.-organizations-providing-high-performance-computing-resources-for-bioinformatics-and-life-sciences"><strong>Table 1.</strong> Organizations providing High-Performance Computing Resources for Bioinformatics and Life Sciences</h3>
<table>
<colgroup>
<col style="width: 40%" />
<col style="width: 60%" />
</colgroup>
<thead>
<tr class="header">
<th>Organization</th>
<th>Computing Resources</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Compute Canada</strong></td>
<td>Provides high-performance computing resources and support services to researchers and innovators across Canada. They offer supercomputers, cloud platforms, data storage, and training programs to advance scientific research and innovation in various fields.</td>
</tr>
<tr class="even">
<td><strong>New York State’s High-Performance Computing Program</strong></td>
<td>Provides researchers, businesses, and educational institutions with access to high-performance computing (HPC) resources and expertise to support their computational research and development efforts.</td>
</tr>
<tr class="odd">
<td><strong>The European Technology Platform for High-Performance Computing</strong></td>
<td>Fosters collaboration among industry, research, and academic stakeholders to advance high-performance computing (HPC) technology in Europe.</td>
</tr>
<tr class="even">
<td><strong>National Center for High-Performance Computing</strong></td>
<td>Facility for high-performance computing (HPC) resources including large-scale computational science and engineering, cluster and grid computing, middleware development, visualization and virtual reality, data storage, networking, and HPC-related training.</td>
</tr>
<tr class="odd">
<td><strong>National Center for Supercomputing Applications</strong></td>
<td>Offers high-performance computing resources such as the Blue Waters supercomputer, provides advanced data storage solutions, data analysis, and visualization tools, and supports interdisciplinary research in fields such as astrophysics, climate modeling, and genomics.</td>
</tr>
<tr class="even">
<td><strong>Oak Ridge Leadership Computing Facility</strong></td>
<td>Provides supercomputing resources, such as the Summit supercomputer, for scientific research, offers support services including software development, data storage, and visualization, and facilitates research in various fields including climate science, biology, and materials science.</td>
</tr>
<tr class="odd">
<td><strong>Swiss National Supercomputing Centre</strong></td>
<td>Provides high-performance computing systems including the Piz Daint supercomputer, offers cloud computing services, data management, and user support, and facilitates scientific research in areas such as climate modeling, physics, and life sciences.</td>
</tr>
<tr class="even">
<td><strong>Barcelona Supercomputing Center</strong></td>
<td>Provides access to MareNostrum, one of the most powerful supercomputers in Europe, offers resources for high-performance computing, data storage, and computational sciences, and supports research in fields including bioinformatics, computational biology, and engineering.</td>
</tr>
<tr class="odd">
<td><strong>Japan’s RIKEN Center for Computational Science</strong></td>
<td>Houses the Fugaku supercomputer, one of the world’s fastest supercomputers, provides resources for computational science, data processing, and artificial intelligence, and supports research in fields such as life sciences, materials science, and disaster prevention.</td>
</tr>
<tr class="even">
<td><strong>National Supercomputing Centre Singapore</strong></td>
<td>Provides high-performance computing resources and support services, offers data storage, cloud computing, and software development services, and supports research in fields including bioinformatics, environmental modeling, and smart cities.</td>
</tr>
</tbody>
</table>
<p>Community computing platforms democratized participation and expanded bioinformatics research’s reach. Platforms like BOINC <span class="citation" data-cites="1BWjfloX3">[<a href="#ref-1BWjfloX3" role="doc-biblioref">49</a>]</span> enabled broad participation in bioinformatics. Experts can submit computing tasks to BOINC, while non-experts and science enthusiasts can volunteer their computer resources to process these tasks. Several life sciences projects are available through BOINC, including protein-ligand docking, malaria simulations, and protein folding <span class="citation" data-cites="1BWjfloX3">[<a href="#ref-1BWjfloX3" role="doc-biblioref">49</a>]</span>.</p>
<h2 id="today-the-present-and-future">2010-Today: The present and future</h2>
<p>The integration of computers into biology has ushered in a new era of research possibilities, allowing for increasingly complex studies. While before, the focus was on individual genes or proteins, advancements today enable the analysis of entire genomes or proteomes <span class="citation" data-cites="1DngqA2eE">[<a href="#ref-1DngqA2eE" role="doc-biblioref">50</a>]</span>. This shift toward a holistic approach in biology is evident in disciplines like genomics, proteomics, and glycomics, which have limited interconnection between them.</p>
<p>The next leap at the intersection of computing and the life sciences lies in modeling entire living organisms and their environments simultaneously, integrating all molecular categories. This has already been achieved in a whole cell model of Mycoplasma genitalium, in which all its genes, products and their known metabolic interactions have been reconstructed <span class="citation" data-cites="FWkpcKEh">[<a href="#ref-FWkpcKEh" role="doc-biblioref">51</a>]</span>. Driven by advancements in measurement techniques, improved computational performance and artificial intelligence (AI) techniques, whole-cell modeling is increasingly becoming realistic and feasible. In contrast to traditional bottom-up approaches relying on molecular interaction networks, a predictive model has been developed for genome-wide phenotypes of budding yeast using deep learning <span class="citation" data-cites="MNG16ntb">[<a href="#ref-MNG16ntb" role="doc-biblioref">52</a>]</span>. The main applications of whole-cell modeling have been in producing useful substances and discovering drugs, such as antimicrobials <span class="citation" data-cites="ZSOb6iY1 EOFcwWBm OHHpDn0z 16yino3ku">[<a href="#ref-ZSOb6iY1" role="doc-biblioref">53</a>,<a href="#ref-EOFcwWBm" role="doc-biblioref">54</a>,<a href="#ref-OHHpDn0z" role="doc-biblioref">55</a>,<a href="#ref-16yino3ku" role="doc-biblioref">56</a>]</span> since whole-cell modeling was first directed toward unicellular organisms. Meanwhile, models of cultured human cells have also been developed, which have found applications in cell differentiation and medical research <span class="citation" data-cites="MgRWKCpM">[<a href="#ref-MgRWKCpM" role="doc-biblioref">57</a>]</span>. The possibility of modeling entire multicellular organisms may not be far off, considering the rapid pace of technological and computational advancements like artificial intelligence (AI) .</p>
<h1 id="advances-in-generative-ai-for-proteins">Advances in Generative AI for Proteins</h1>
<h2 id="introduction-1">Introduction</h2>
<ul>
<li>Background on proteins as key molecular machines in biology</li>
<li>Promise of generative AI to accelerate protein discovery and engineering</li>
<li>Overview of scope covering recent advances in last 1-2 years</li>
</ul>
<h2 id="structure-prediction">Structure Prediction</h2>
<ul>
<li>AlphaFold2 as a breakthrough method for structure prediction</li>
<li>Novel model architecture and training methodology</li>
<li>Examples of new biological insights from predicted structures</li>
</ul>
<h2 id="function-prediction">Function Prediction</h2>
<ul>
<li>Using predicted structures to infer protein functions</li>
<li>Structure-based identification of catalytic and binding sites</li>
<li>Case studies of novel enzyme functions discovered</li>
</ul>
<h2 id="designing-novel-proteins">Designing Novel Proteins</h2>
<ul>
<li>Generative models for designing functional protein sequences</li>
<li>Leveraging structural constraints for optimized protein engineering</li>
<li>Applications in industrial enzymes, therapeutics, biomaterials</li>
</ul>
<h2 id="interaction-prediction">Interaction Prediction</h2>
<ul>
<li>Modeling protein-protein interactions with graph networks</li>
<li>Structure-based prediction of protein-drug bindings</li>
<li>Applications in drug discovery and toxicity screening</li>
</ul>
<h2 id="outlook">Outlook</h2>
<ul>
<li>Challenges and next steps in improving accuracy</li>
<li>Hybrid physics- and data-driven approaches</li>
<li>Ethical considerations in synthetic protein design</li>
</ul>
<h1 id="advances-in-generative-ai-for-rna">Advances in Generative AI for RNA</h1>
<h2 id="introduction-2">Introduction</h2>
<ul>
<li>Background on key roles of RNA in biology</li>
<li>Promise of generative models to advance RNA research</li>
<li>Scope focused on latest advances in RNA prediction and design</li>
</ul>
<h2 id="structure-prediction-1">Structure Prediction</h2>
<ul>
<li>Transformer-based prediction of RNA folding</li>
<li>Novel architectures for incorporating chemical constraints</li>
<li>Improved accuracy on complex structures like ribosomes</li>
</ul>
<h2 id="function-prediction-1">Function Prediction</h2>
<ul>
<li>Inferring RNA functions from sequence and structure</li>
<li>Identifying motifs, domains, and atomic binding sites</li>
<li>Applications in understanding long noncoding RNAs</li>
</ul>
<h2 id="interaction-prediction-1">Interaction Prediction</h2>
<ul>
<li>Graph neural networks for RNA-protein interactions</li>
<li>Structure-augmented modeling of splice sites</li>
<li>Predicting RNA base editing targets</li>
</ul>
<h2 id="design-of-rna-therapeutics">Design of RNA Therapeutics</h2>
<ul>
<li>Generative models for optimizing siRNA, antisense design</li>
<li>Reinforcement learning for chemical modification patterns</li>
<li>Progress in computational RNA-targeted drug design</li>
</ul>
<h2 id="outlook-1">Outlook</h2>
<ul>
<li>Key challenges in prediction of long RNA structures</li>
<li>Design of RNA for self-assembly and scaffolding</li>
<li>Ethical use of synthetic RNA technologies</li>
</ul>
<h1 id="advances-in-generative-ai-for-dna">Advances in Generative AI for DNA</h1>
<h2 id="introduction-3">Introduction</h2>
<ul>
<li>Background on the role of DNA as a carrier of genetic information<br />
</li>
<li>Promise of generative models to advance DNA research</li>
<li>Scope focused on the latest advances in DNA prediction and design</li>
</ul>
<h2 id="sequence-modeling">Sequence Modeling</h2>
<h3 id="transformer-architectures-for-modeling-dna-sequences">Transformer architectures for modeling DNA sequences</h3>
<p>The HyenaDNA paper introduces a new genomic foundation model called HyenaDNA that can process DNA sequences at single nucleotide resolution with ultralong context lengths up to 1 million base pairs - a 500x increase over previous transformer models. HyenaDNA uses a parameter-efficient convolutional architecture that allows it to scale subquadratically with sequence length, enabling the use of full genome-scale context. On a range of regulatory genomics prediction tasks, HyenaDNA matches or exceeds the performance of previous state-of-the-art models while using 1500x fewer parameters and 3200x less pretraining data. HyenaDNA also demonstrates the ability to perform challenging species classification using the full mutational profile visible at 1 million base pairs of context. The authors explore new training techniques to enable ultralong sequence modeling as well as prompt-based tuning methods for rapidly adapting to new tasks without updating pre-trained weights <span class="citation" data-cites="3wB77oTL">[<a href="#ref-3wB77oTL" role="doc-biblioref">58</a>]</span>.</p>
<h3 id="pretraining-on-large-genomic-datasets">Pretraining on large genomic datasets</h3>
<h3 id="applications-in-variant-calling-and-annotation">Applications in variant calling and annotation</h3>
<h2 id="regulation-prediction">Regulation Prediction</h2>
<ul>
<li>Graph neural networks for modeling 3D genome architecture</li>
<li>Predicting enhancer-promoter interactions and expression</li>
<li>Design of synthetic promoters and enhancers</li>
</ul>
<h2 id="genome-editing">Genome Editing</h2>
<ul>
<li>Generative models for CRISPR guide design</li>
<li>Contextual prediction of on-target editing efficacy<br />
</li>
<li>Modeling of off-target effects during optimization</li>
</ul>
<h2 id="dna-data-generation">DNA Data Generation</h2>
<ul>
<li>Variational autoencoders for realistic DNA sequences</li>
<li>Generating paired genomic-transcriptomic data</li>
<li>Applications in training genome interpretation models</li>
</ul>
<h2 id="outlook-2">Outlook</h2>
<h3 id="challenges-in-predicting-long-range-chromatin-interactions">Challenges in predicting long-range chromatin interactions</h3>
<p>The development of transformer architectures like HyenaDNA that can leverage genome-scale context creates new opportunities for understanding long-range chromatin interactions, gene regulation, and intercellular networks. However, significant challenges remain in improving model accuracy and uncertainty quantification. Hybrid physics- and data-driven approaches may help address these gaps. Safety considerations around synthetic genome design also warrant further research to ensure responsible innovation as these generative capabilities advance. Overall, the future looks bright for generative models that can capture both local mutations and global patterns critical to biology <span class="citation" data-cites="3wB77oTL">[<a href="#ref-3wB77oTL" role="doc-biblioref">58</a>]</span>.</p>
<h3 id="responsible-design-of-synthetic-genomes">Responsible design of synthetic genomes</h3>
<h3 id="ethical-considerations-for-human-genome-editing">Ethical considerations for human genome editing</h3>
<h1 id="generative-ai-for-autonomous-experimentation">Generative AI for Autonomous Experimentation</h1>
<h2 id="introduction-4">Introduction</h2>
<ul>
<li>Promise of generative models to accelerate scientific discovery</li>
<li>Rise of autonomous labs and robotics for automated experimentation</li>
<li>Overview of generative AI’s role in self-driving research</li>
</ul>
<h2 id="closed-loop-systems">Closed-Loop Systems</h2>
<ul>
<li>Integrating computational hypothesis generation with robotic wet lab testing</li>
<li>Reinforcement learning pipelines for autonomous optimization</li>
<li>Case studies in materials science, drug discovery</li>
</ul>
<h2 id="automated-experiment-design">Automated Experiment Design</h2>
<ul>
<li>Using generative models to design novel compounds, genes</li>
<li>Leveraging simulations to predict experimental outcomes</li>
<li>Robotic execution of designed experiments</li>
</ul>
<h2 id="adaptive-sampling">Adaptive Sampling</h2>
<ul>
<li>Active learning to iteratively select most informative experiments</li>
<li>Bayesian optimization powered by neural networks</li>
<li>Applications in probing molecular design spaces</li>
</ul>
<h2 id="real-time-learning">Real-Time Learning</h2>
<ul>
<li>Deploying models on lab edge devices</li>
<li>Online learning from experimental data streams</li>
<li>Improving models and experiment plans on-the-fly</li>
</ul>
<h2 id="outlook-3">Outlook</h2>
<ul>
<li>Key challenges around model accuracy and integration</li>
<li>The future of data-driven, self-driving laboratories</li>
<li>Risks of full automation and need for human oversight</li>
</ul>
<h2 id="conclusion">Conclusion</h2>
<ul>
<li>Generative AI as a powerful tool for autonomous experimentation</li>
<li>Accelerating discovery alongside human researchers</li>
<li>Responsible implementation will maximize benefit</li>
</ul>
<h1 id="generative-ai-and-the-biosecurity-landscape">Generative AI and the Biosecurity Landscape</h1>
<h2 id="introduction-5">Introduction</h2>
<ul>
<li>Background on biosecurity threats from natural, accidental, and intentional pathogens</li>
<li>Rise of generative AI as a dual use technology for biodefense and misuse</li>
</ul>
<h2 id="enhanced-risks">Enhanced Risks</h2>
<ul>
<li>Automated bioweapon design with generative models</li>
<li>Relatively low computing needs to generate dangerous agents</li>
<li>Challenges detecting artificially generated sequences/organisms</li>
</ul>
<h2 id="enhanced-response-capabilities">Enhanced Response Capabilities</h2>
<ul>
<li>Generative models for vaccine and therapeutic design</li>
<li>High-throughput testing of countermeasures with synthetic data</li>
<li>AI for early detection of emergent pathogens and outbreaks</li>
</ul>
<h2 id="recommendations">Recommendations</h2>
<ul>
<li>Increased oversight for generative model development/release</li>
<li>Expanding biosecurity legislation and regulations</li>
<li>Fostering open research and global cooperation</li>
</ul>
<h2 id="outlook-4">Outlook</h2>
<ul>
<li>Trajectory toward increasingly powerful generative biological capabilities<br />
</li>
<li>Need for preventative ethics research and guidance</li>
<li>Maintaining public trust and avoiding overreaction</li>
</ul>
<h2 id="conclusion-1">Conclusion</h2>
<ul>
<li>Balancing generative AI’s benefits and risks in biology</li>
<li>Importance of thoughtful governance and responsible innovation</li>
<li>Staying ahead of the curve on biosecurity</li>
</ul>
<h1 id="policy-responses-to-generative-ai-in-biology">Policy Responses to Generative AI in Biology</h1>
<h2 id="introduction-6">Introduction</h2>
<ul>
<li>Background on rise of powerful generative models for biology</li>
<li>Overview of risks like bioweapons, environmental damage</li>
<li>Need for governance to ensure responsible development</li>
</ul>
<h2 id="self-governance-by-developers">Self-Governance by Developers</h2>
<ul>
<li>Voluntary guidelines on ethical AI by corporations<br />
</li>
<li>Limiting access to certain capabilities like human editing</li>
<li>Issues with self-regulation and transparency</li>
</ul>
<h2 id="governmental-regulations">Governmental Regulations</h2>
<ul>
<li>New biosecurity regulations on certain AI technologies</li>
<li>Restrictions on use of synthetic biology IP</li>
<li>Challenges with fast pace of technology change</li>
</ul>
<h2 id="international-governance">International Governance</h2>
<ul>
<li>Proposals for global observatory to monitor risks</li>
<li>Treaties restricting development of bioweapons<br />
</li>
<li>Difficulty achieving consensus and compliance</li>
</ul>
<h2 id="public-deliberation-and-ethics">Public Deliberation and Ethics</h2>
<ul>
<li>Activities to involve broader stakeholders</li>
<li>Gathering public attitudes on acceptable uses of generative bio AI</li>
<li>Informing policy with deliberative democracy</li>
</ul>
<h2 id="outlook-5">Outlook</h2>
<ul>
<li>Likely increase in debate and policy activity in this space</li>
<li>Balancing innovation and security will be a key challenge</li>
<li>Importance of thoughtful multidisciplinary discourse</li>
</ul>
<h2 id="conclusion-2">Conclusion</h2>
<ul>
<li>No easy policy solutions, but inaction also carries risks</li>
<li>Policy should enable innovation but promote responsible use</li>
<li>This will require sustained public deliberation and coordination</li>
</ul>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-gldEpQ9" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Computers in the study of evolution</strong> <div class="csl-block">JL Crosby</div> <em>Science Progress</em> (1967) <a href="https://pubmed.ncbi.nlm.nih.gov/4859964">https://pubmed.ncbi.nlm.nih.gov/4859964</a></div>
</div>
<div id="ref-UK6rEYVY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>The chemical basis of morphogenesis</strong> <div class="csl-block">Philosophical Transactions of the Royal Society of London. Series B, Biological Sciences</div> (1952-08-14) <a href="https://royalsocietypublishing.org/doi/10.1098/rstb.1952.0012">https://royalsocietypublishing.org/doi/10.1098/rstb.1952.0012</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1098/rstb.1952.0012">10.1098/rstb.1952.0012</a></div></div>
</div>
<div id="ref-RqzQpRL0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Scientific uses of the MANIAC</strong> <div class="csl-block">HL Anderson</div> <em>Journal of Statistical Physics</em> (1986-06-01) <a href="https://doi.org/10.1007/BF02628301">https://doi.org/10.1007/BF02628301</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/BF02628301">10.1007/bf02628301</a></div></div>
</div>
<div id="ref-AtEWfwCG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>A three-dimensional model of the myoglobin molecule obtained by x-ray analysis</strong> <div class="csl-block">JC Kendrew, G Bodo, HM Dintzis, RG Parrish, H Wyckoff, DC Phillips</div> <em>Nature</em> (1958-03-08) <a href="https://pubmed.ncbi.nlm.nih.gov/13517261">https://pubmed.ncbi.nlm.nih.gov/13517261</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/181662a0">10.1038/181662a0</a></div></div>
</div>
<div id="ref-WmbDqOWv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>Structure of myoglobin: A three-dimensional Fourier synthesis at 2 A. resolution</strong> <div class="csl-block">JC Kendrew, RE Dickerson, BE Strandberg, RG Hart, DR Davies, DC Phillips, VC Shore</div> <em>Nature</em> (1960-02-13) <a href="https://pubmed.ncbi.nlm.nih.gov/18990802">https://pubmed.ncbi.nlm.nih.gov/18990802</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/185422a0">10.1038/185422a0</a></div></div>
</div>
<div id="ref-3t4YHcSd" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>The amino-acid sequence in the glycyl chain of insulin. 1. The identification of lower peptides from partial hydrolysates</strong> <div class="csl-block">F Sanger, EOP Thompson</div> <em>Biochemical Journal</em> (1953-02) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1198157/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1198157/</a></div>
</div>
<div id="ref-Htmw8erk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>The amino-acid sequence in the glycyl chain of insulin. 2. The investigation of peptides from enzymic hydrolysates</strong> <div class="csl-block">F Sanger, EOP Thompson</div> <em>Biochemical Journal</em> (1953-02) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1198158/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1198158/</a></div>
</div>
<div id="ref-Dq9jMQ6v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>A method for the determination of amino acid sequence in peptides</strong> <div class="csl-block">P Edman</div> <em>Archives of Biochemistry</em> (1949-07) <a href="https://pubmed.ncbi.nlm.nih.gov/18134557">https://pubmed.ncbi.nlm.nih.gov/18134557</a></div>
</div>
<div id="ref-prCR1hpf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>The origins of bioinformatics</strong> <div class="csl-block">JB Hagen</div> <em>Nature Reviews. Genetics</em> (2000-12) <a href="https://pubmed.ncbi.nlm.nih.gov/11252753">https://pubmed.ncbi.nlm.nih.gov/11252753</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/35042090">10.1038/35042090</a></div></div>
</div>
<div id="ref-1Bxyvu5tQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Comprotein: a computer program to aid primary protein structure determination</strong> <div class="csl-block">Margaret Oakley Dayhoff, Robert S Ledley</div> <em>Proceedings of the December 4-6, 1962, fall joint computer conference on - AFIPS '62 (Fall)</em> (1962) <a href="http://portal.acm.org/citation.cfm?doid=1461518.1461546">http://portal.acm.org/citation.cfm?doid=1461518.1461546</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1145/1461518.1461546">10.1145/1461518.1461546</a></div></div>
</div>
<div id="ref-ibvKtnrl" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><a href="https://febs.onlinelibrary.wiley.com/toc/14321033/5/2">https://febs.onlinelibrary.wiley.com/toc/14321033/5/2</a></div>
</div>
<div id="ref-YYcdJkZI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Collecting, comparing, and computing sequences: the making of Margaret O. Dayhoff's Atlas of Protein Sequence and Structure, 1954-1965</strong> <div class="csl-block">Bruno J Strasser</div> <em>Journal of the History of Biology</em> (2010) <a href="https://pubmed.ncbi.nlm.nih.gov/20665074">https://pubmed.ncbi.nlm.nih.gov/20665074</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10739-009-9221-0">10.1007/s10739-009-9221-0</a></div></div>
</div>
<div id="ref-C1oThCRN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>A general method applicable to the search for similarities in the amino acid sequence of two proteins</strong> <div class="csl-block">SB Needleman, CD Wunsch</div> <em>Journal of Molecular Biology</em> (1970-03) <a href="https://pubmed.ncbi.nlm.nih.gov/5420325">https://pubmed.ncbi.nlm.nih.gov/5420325</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/0022-2836(70)90057-4">10.1016/0022-2836(70)90057-4</a></div></div>
</div>
<div id="ref-9IbNBDhy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Progressive sequence alignment as a prerequisite to correct phylogenetic trees</strong> <div class="csl-block">DF Feng, RF Doolittle</div> <em>Journal of Molecular Evolution</em> (1987) <a href="https://pubmed.ncbi.nlm.nih.gov/3118049">https://pubmed.ncbi.nlm.nih.gov/3118049</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/BF02603120">10.1007/bf02603120</a></div></div>
</div>
<div id="ref-r9luxm5X" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>CLUSTAL: a package for performing multiple sequence alignment on a microcomputer</strong> <div class="csl-block">DG Higgins, PM Sharp</div> <em>Gene</em> (1988-12-15) <a href="https://pubmed.ncbi.nlm.nih.gov/3243435">https://pubmed.ncbi.nlm.nih.gov/3243435</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/0378-1119(88)90330-7">10.1016/0378-1119(88)90330-7</a></div></div>
</div>
<div id="ref-10odPMDxs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Clustal Omega, accurate alignment of very large numbers of sequences</strong> <div class="csl-block">Fabian Sievers, Desmond G Higgins</div> <em>Methods in Molecular Biology (Clifton, N.J.)</em> (2014) <a href="https://pubmed.ncbi.nlm.nih.gov/24170397">https://pubmed.ncbi.nlm.nih.gov/24170397</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-1-62703-646-7_6">10.1007/978-1-62703-646-7_6</a></div></div>
</div>
<div id="ref-NQbXc701" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>A new method for sequencing DNA</strong> <div class="csl-block">AM Maxam, W Gilbert</div> <em>Proceedings of the National Academy of Sciences of the United States of America</em> (1977-02) <a href="https://pubmed.ncbi.nlm.nih.gov/265521">https://pubmed.ncbi.nlm.nih.gov/265521</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.74.2.560">10.1073/pnas.74.2.560</a></div></div>
</div>
<div id="ref-10pibmvLC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Summary statement of the Asilomar conference on recombinant DNA molecules.</strong> <div class="csl-block">P Berg, D Baltimore, S Brenner, RO Roblin, MF Singer</div> <em>Proceedings of the National Academy of Sciences of the United States of America</em> (1975-06) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC432675/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC432675/</a></div>
</div>
<div id="ref-M8aKGlSe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>A strategy of DNA sequencing employing computer programs.</strong> <div class="csl-block">R Staden</div> <em>Nucleic Acids Research</em> (1979-06-11) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC327874/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC327874/</a></div>
</div>
<div id="ref-nzs4mjEF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Evolutionary trees from DNA sequences: a maximum likelihood approach</strong> <div class="csl-block">J Felsenstein</div> <em>Journal of Molecular Evolution</em> (1981) <a href="https://pubmed.ncbi.nlm.nih.gov/7288891">https://pubmed.ncbi.nlm.nih.gov/7288891</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/BF01734359">10.1007/bf01734359</a></div></div>
</div>
<div id="ref-1CARqSM3b" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Probability distribution of molecular evolutionary trees: a new method of phylogenetic inference</strong> <div class="csl-block">B Rannala, Z Yang</div> <em>Journal of Molecular Evolution</em> (1996-09) <a href="https://pubmed.ncbi.nlm.nih.gov/8703097">https://pubmed.ncbi.nlm.nih.gov/8703097</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/BF02338839">10.1007/bf02338839</a></div></div>
</div>
<div id="ref-17nhyKsKm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>A biologist's guide to Bayesian phylogenetic analysis</strong> <div class="csl-block">Fabrícia F Nascimento, Mario Dos Reis, Ziheng Yang</div> <em>Nature Ecology &amp; Evolution</em> (2017-10) <a href="https://pubmed.ncbi.nlm.nih.gov/28983516">https://pubmed.ncbi.nlm.nih.gov/28983516</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41559-017-0280-x">10.1038/s41559-017-0280-x</a></div></div>
</div>
<div id="ref-CvVpaf4a" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>The Nobel Prize in Chemistry 1993</strong> <div class="csl-block">NobelPrize.org</div> <a href="https://www.nobelprize.org/prizes/chemistry/1993/mullis/lecture/">https://www.nobelprize.org/prizes/chemistry/1993/mullis/lecture/</a></div>
</div>
<div id="ref-12FKql4zv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>A comprehensive set of sequence analysis programs for the VAX</strong> <div class="csl-block">J Devereux, P Haeberli, O Smithies</div> <em>Nucleic Acids Research</em> (1984-01-11) <a href="https://pubmed.ncbi.nlm.nih.gov/6546423">https://pubmed.ncbi.nlm.nih.gov/6546423</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/12.1part1.387">10.1093/nar/12.1part1.387</a></div></div>
</div>
<div id="ref-17EunIaqv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>DNASTAR’s Lasergene Sequence Analysis Software</strong> <div class="csl-block">Timothy G Burland</div> <em>Bioinformatics Methods and Protocols</em> (1999) <a href="https://doi.org/10.1385/1-59259-192-2:71">https://doi.org/10.1385/1-59259-192-2:71</a> <div class="csl-block">ISBN: 9781592591923</div></div>
</div>
<div id="ref-dSNqAXK0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>Apple II PASCAL programs for molecular biologists</strong> <div class="csl-block">B Malthiery, B Bellon, D Giorgi, B Jacq</div> <em>Nucleic Acids Research</em> (1984-01-11) <a href="https://pubmed.ncbi.nlm.nih.gov/6320099">https://pubmed.ncbi.nlm.nih.gov/6320099</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/12.1part2.569">10.1093/nar/12.1part2.569</a></div></div>
</div>
<div id="ref-17gPkdn86" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>The GNU Manifesto - GNU Project - Free Software Foundation</strong> <a href="https://www.gnu.org/gnu/manifesto.en.html">https://www.gnu.org/gnu/manifesto.en.html</a></div>
</div>
<div id="ref-WD5AIwcy" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><a href="https://www.researchgate.net/publication/221307757_The_Free_Software_Movement_and_the_GNULinux_Operating_System">https://www.researchgate.net/publication/221307757_The_Free_Software_Movement_and_the_GNULinux_Operating_System</a></div>
</div>
<div id="ref-11GoDnHp3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Whole-genome random sequencing and assembly of Haemophilus influenzae Rd</strong> <div class="csl-block">RD Fleischmann, MD Adams, O White, RA Clayton, EF Kirkness, AR Kerlavage, CJ Bult, JF Tomb, BA Dougherty, JM Merrick</div> <em>Science (New York, N.Y.)</em> (1995-07-28) <a href="https://pubmed.ncbi.nlm.nih.gov/7542800">https://pubmed.ncbi.nlm.nih.gov/7542800</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.7542800">10.1126/science.7542800</a></div></div>
</div>
<div id="ref-7GkBaAWF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>The sequence of the human genome</strong> <div class="csl-block">JC Venter, MD Adams, EW Myers, PW Li, RJ Mural, GG Sutton, HO Smith, M Yandell, CA Evans, RA Holt, … X Zhu</div> <em>Science (New York, N.Y.)</em> (2001-02-16) <a href="https://pubmed.ncbi.nlm.nih.gov/11181995">https://pubmed.ncbi.nlm.nih.gov/11181995</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1058040">10.1126/science.1058040</a></div></div>
</div>
<div id="ref-Ero6Muzo" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Consed: a graphical tool for sequence finishing</strong> <div class="csl-block">D Gordon, C Abajian, P Green</div> <em>Genome Research</em> (1998-03) <a href="https://pubmed.ncbi.nlm.nih.gov/9521923">https://pubmed.ncbi.nlm.nih.gov/9521923</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.8.3.195">10.1101/gr.8.3.195</a></div></div>
</div>
<div id="ref-18guJvxxN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>A whole-genome assembly of Drosophila</strong> <div class="csl-block">EW Myers, GG Sutton, AL Delcher, IM Dew, DP Fasulo, MJ Flanigan, SA Kravitz, CM Mobarry, KH Reinert, KA Remington, … JC Venter</div> <em>Science (New York, N.Y.)</em> (2000-03-24) <a href="https://pubmed.ncbi.nlm.nih.gov/10731133">https://pubmed.ncbi.nlm.nih.gov/10731133</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.287.5461.2196">10.1126/science.287.5461.2196</a></div></div>
</div>
<div id="ref-i2bVokif" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>The EMBL data library</strong> <div class="csl-block">CM Rice, R Fuchs, DG Higgins, PJ Stoehr, GN Cameron</div> <em>Nucleic Acids Research</em> (1993-07-01) <a href="https://pubmed.ncbi.nlm.nih.gov/8332519">https://pubmed.ncbi.nlm.nih.gov/8332519</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/21.13.2967">10.1093/nar/21.13.2967</a></div></div>
</div>
<div id="ref-Tx5CQPm8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>GenBank</strong> <div class="csl-block">Dennis A Benson, Mark Cavanaugh, Karen Clark, Ilene Karsch-Mizrachi, David J Lipman, James Ostell, Eric W Sayers</div> <em>Nucleic Acids Research</em> (2013-01) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3531190/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3531190/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gks1195">10.1093/nar/gks1195</a></div></div>
</div>
<div id="ref-qPG24gP6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>BLAST: at the core of a powerful and diverse set of sequence analysis tools</strong> <div class="csl-block">Scott McGinnis, Thomas L Madden</div> <em>Nucleic Acids Research</em> (2004-07-01) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC441573/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC441573/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkh435">10.1093/nar/gkh435</a></div></div>
</div>
<div id="ref-14cnpLsxs" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Genomes OnLine Database (GOLD):
a monitor of genome projects world-wide</strong> <div class="csl-block">Axel Bernal, Uy Ear, Nikos Kyrpides</div> <em>Nucleic Acids Research</em> (2001-01-01) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC29859/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC29859/</a></div>
</div>
<div id="ref-1DNwvIpHH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>PubMed</strong> <div class="csl-block">PubMed</div> <a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a></div>
</div>
<div id="ref-yOdF9F1J" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><a href="https://academic.oup.com/nar/article/26/1/94/2379498">https://academic.oup.com/nar/article/26/1/94/2379498</a></div>
</div>
<div id="ref-MDJLJlU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><a href="https://Abramson">Abramson</a></div>
</div>
<div id="ref-UglDtG3f" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>The BLEND system Programme for the study of some ‘electronic journals’∗</strong> <div class="csl-block">B Shackel</div> <em>Ergonomics</em> (1982-04) <a href="http://www.tandfonline.com/doi/abs/10.1080/00140138208924954">http://www.tandfonline.com/doi/abs/10.1080/00140138208924954</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1080/00140138208924954">10.1080/00140138208924954</a></div></div>
</div>
<div id="ref-jMSKwR4o" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>arXiv.org e-Print archive</strong> <a href="https://arxiv.org/">https://arxiv.org/</a></div>
</div>
<div id="ref-iWYRgRGv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>bioRxiv.org - the preprint server for Biology</strong> <a href="https://www.biorxiv.org/">https://www.biorxiv.org/</a></div>
</div>
<div id="ref-4gIVWSCR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Configurations of Polypeptide Chains With Favored Orientations Around Single Bonds: Two New Pleated Sheets</strong> <div class="csl-block">L Pauling, RB Corey</div> <em>Proceedings of the National Academy of Sciences of the United States of America</em> (1951-11) <a href="https://pubmed.ncbi.nlm.nih.gov/16578412">https://pubmed.ncbi.nlm.nih.gov/16578412</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1073/pnas.37.11.729">10.1073/pnas.37.11.729</a></div></div>
</div>
<div id="ref-cc2nygki" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Sixty-five years of the long march in protein secondary structure prediction: the final stretch?</strong> <div class="csl-block">Yuedong Yang, Jianzhao Gao, Jihua Wang, Rhys Heffernan, Jack Hanson, Kuldip Paliwal, Yaoqi Zhou</div> <em>Briefings in Bioinformatics</em> (2018-05-01) <a href="https://pubmed.ncbi.nlm.nih.gov/28040746">https://pubmed.ncbi.nlm.nih.gov/28040746</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbw129">10.1093/bib/bbw129</a></div></div>
</div>
<div id="ref-eM1kmLmc" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Computational methods for protein structure prediction and modeling</strong> <div class="csl-block">New York, N.Y. : Springer</div> (2007) <a href="http://archive.org/details/computationalmet0000unse_u4q5">http://archive.org/details/computationalmet0000unse_u4q5</a> <div class="csl-block">ISBN: 9780387333212</div></div>
</div>
<div id="ref-Z7e2T8go" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>Molecular dynamics simulations: advances and applications</strong> <div class="csl-block">Adam Hospital, Josep Ramon Goñi, Modesto Orozco, Josep L Gelpí</div> <em>Advances and Applications in Bioinformatics and Chemistry : AABC</em> (2015-11-19) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4655909/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4655909/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.2147/AABC.S70333">10.2147/aabc.s70333</a></div></div>
</div>
<div id="ref-sT4CEJLS" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>To Milliseconds and Beyond: Challenges in the Simulation of Protein Folding</strong> <div class="csl-block">Thomas J Lane, Diwakar Shukla, Kyle A Beauchamp, Vijay S Pande</div> <em>Current opinion in structural biology</em> (2013-02) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3673555/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3673555/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.sbi.2012.11.002">10.1016/j.sbi.2012.11.002</a></div></div>
</div>
<div id="ref-16JwdMPBH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>Next-Generation Sequencing and Its Application: Empowering in Public Health Beyond Reality</strong> <div class="csl-block">Nidhi Gupta, Vijay K Verma</div> <em>Microbial Technology for the Welfare of Society</em> (2019-09-13) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7122948/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7122948/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-981-13-8844-6_15">10.1007/978-981-13-8844-6_15</a></div></div>
</div>
<div id="ref-1BWjfloX3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>BOINC: A Platform for Volunteer Computing</strong> <div class="csl-block">David P Anderson</div> <em>Journal of Grid Computing</em> (2020-03-01) <a href="https://doi.org/10.1007/s10723-019-09497-9">https://doi.org/10.1007/s10723-019-09497-9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10723-019-09497-9">10.1007/s10723-019-09497-9</a></div></div>
</div>
<div id="ref-1DngqA2eE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>Structural proteomics by NMR spectroscopy</strong> <div class="csl-block">Joon Shin, Woonghee Lee, Weontae Lee</div> <em>Expert Review of Proteomics</em> (2008-08) <a href="https://pubmed.ncbi.nlm.nih.gov/18761469">https://pubmed.ncbi.nlm.nih.gov/18761469</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1586/14789450.5.4.589">10.1586/14789450.5.4.589</a></div></div>
</div>
<div id="ref-FWkpcKEh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>A whole-cell computational model predicts phenotype from genotype</strong> <div class="csl-block">Jonathan R Karr, Jayodita C Sanghvi, Derek N Macklin, Miriam V Gutschow, Jared M Jacobs, Benjamin Bolival, Nacyra Assad-Garcia, John I Glass, Markus W Covert</div> <em>Cell</em> (2012-07-20) <a href="https://pubmed.ncbi.nlm.nih.gov/22817898">https://pubmed.ncbi.nlm.nih.gov/22817898</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2012.05.044">10.1016/j.cell.2012.05.044</a></div></div>
</div>
<div id="ref-MNG16ntb" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>Using deep learning to model the hierarchical structure and function of a cell</strong> <div class="csl-block">Jianzhu Ma, Michael Ku Yu, Samson Fong, Keiichiro Ono, Eric Sage, Barry Demchak, Roded Sharan, Trey Ideker</div> <em>Nature Methods</em> (2018-04) <a href="https://www.nature.com/articles/nmeth.4627">https://www.nature.com/articles/nmeth.4627</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.4627">10.1038/nmeth.4627</a></div></div>
</div>
<div id="ref-ZSOb6iY1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>Why Build Whole-Cell Models?</strong> <div class="csl-block">Javier Carrera, Markus W Covert</div> <em>Trends in Cell Biology</em> (2015-12) <a href="https://pubmed.ncbi.nlm.nih.gov/26471224">https://pubmed.ncbi.nlm.nih.gov/26471224</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.tcb.2015.09.004">10.1016/j.tcb.2015.09.004</a></div></div>
</div>
<div id="ref-EOFcwWBm" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>The future of whole-cell modeling</strong> <div class="csl-block">Derek N Macklin, Nicholas A Ruggero, Markus W Covert</div> <em>Current Opinion in Biotechnology</em> (2014-08) <a href="https://pubmed.ncbi.nlm.nih.gov/24556244">https://pubmed.ncbi.nlm.nih.gov/24556244</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.copbio.2014.01.012">10.1016/j.copbio.2014.01.012</a></div></div>
</div>
<div id="ref-OHHpDn0z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>Computer-Aided Whole-Cell Design: Taking a Holistic Approach by Integrating Synthetic With Systems Biology</strong> <div class="csl-block">Lucia Marucci, Matteo Barberis, Jonathan Karr, Oliver Ray, Paul R Race, Miguel de Souza Andrade, Claire Grierson, Stefan Andreas Hoffmann, Sophie Landon, Elibio Rech, … Christopher Woods</div> <em>Frontiers in Bioengineering and Biotechnology</em> (2020-08-07) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7426639/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7426639/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fbioe.2020.00942">10.3389/fbioe.2020.00942</a></div></div>
</div>
<div id="ref-16yino3ku" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>Accelerated discovery via a whole-cell model</strong> <div class="csl-block">Jayodita C Sanghvi, Sergi Regot, Silvia Carrasco, Jonathan R Karr, Miriam V Gutschow, Benjamin Bolival, Markus W Covert</div> <em>Nature methods</em> (2013-12) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3856890/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3856890/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.2724">10.1038/nmeth.2724</a></div></div>
</div>
<div id="ref-MgRWKCpM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>A blueprint for human whole-cell modeling</strong> <div class="csl-block">Balázs Szigeti, Yosef D Roth, John AP Sekar, Arthur P Goldberg, Saahith C Pochiraju, Jonathan R Karr</div> <em>Current opinion in systems biology</em> (2018-02) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5966287/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5966287/</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.coisb.2017.10.005">10.1016/j.coisb.2017.10.005</a></div></div>
</div>
<div id="ref-3wB77oTL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>HyenaDNA: Long-Range Genomic Sequence Modeling at Single Nucleotide Resolution</strong> <div class="csl-block">Eric Nguyen, Michael Poli, Marjan Faizi, Armin Thomas, Callum Birch-Sykes, Michael Wornow, Aman Patel, Clayton Rabideau, Stefano Massaroli, Yoshua Bengio, … Chris Ré</div> <em>arXiv</em> (2023) <a href="https://doi.org/gs3v5j">https://doi.org/gs3v5j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.48550/arxiv.2306.15794">10.48550/arxiv.2306.15794</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
